app/questionnaire/rules.py

```diff
@@ -62,6 +62,15 @@ def evaluate_condition(condition, answer_value, match_value):
     """
     answer_and_match = answer_value is not None and match_value is not None
 
+    if condition in {"equals", "not equals", "equals any", "not equals any"}:
+
+        answer_value = casefold(answer_value)
```

Comment:
Given we have a lot of routing for census using non str answers like dates, this func would be excepting AttributeError a lot, therefore, would we rather not do:
```py
answer_value = (
            casefold(answer_value) if isinstance(answer_value, str) else answer_value
        )
```
or make this check part of `casefold()`. Same point for `match_value`. Thoughts?

Reply:
Unless `isinstance` can be shown to be qualitatively better in this scenario then I'll keep the duck typing approach and not hard code to the `str` builtin

Reply:
`isinstance()` would be more performant if most cases weren't strs but given it should be negligible I'm happy to leave as is.